src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -59,6 +66,26 @@ public function process(ContainerBuilder $container)
                 $tmpContainer->addObjectResource($configuration);
             }
 
+            if (null !== $envTypes && method_exists(BaseNode::class, 'setPlaceholder')) {
+                $defaultEnvBag = new ParameterBag($parameters);
+                foreach ($resolvingBag->getEnvPlaceholders() as $env => $placeholders) {
+                    $prefix = (false === $i = strpos($env, ':')) ? 'string' : substr($env, 0, $i);
+                    $types = isset($envTypes[$prefix]) ? $envTypes[$prefix] : array('string');
+                    $default = ($hasEnv = (false === $i && $defaultEnvBag->has("env($env)"))) ? $defaultEnvBag->get("env($env)") : null;
+                    if (null !== $default && !in_array($type = gettype($default), $types, true)) {
+                        throw new LogicException(sprintf('Invalid type for env parameter "env(%s)", expected "%s" got "%s".', $env, implode('", "', $types), $type));
+                    }
+                    $values = array();
+                    $fixtures = array('array' => array(), 'bool' => false, 'float' => 0.0, 'int' => 0, 'string' => '');
```

Comment:
Put that as private static? Are the keys correct?

Reply:
:disappointed: forgot to stage. All good now.